src/lib/Libdis/ps_dis.c

```diff
@@ -461,5 +461,5 @@ free_and_return(vnl_t *vnlp)
 	free(vnlp->vnl_list);
 	free(vnlp);
 
-	return ((vnl_t *) NULL);
+	return NULL;
```

Comment:
Just a note about the main loop... while the i'th element may be partially allocated, it may not have been allocated at all. We may want to initialize vnl_cur to -1 in the vn_decode_DIS routines.

Reply:
If vnlp is calloced, then it will all be initialized to 0. We only call `free_and_return` after we start building the list, so vnl_cur doesn't need to be initialized to -1.